indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerConfig.java

```diff
@@ -107,8 +109,10 @@ public void configure(Binder binder)
             new IndexingHadoopModule()
         )
     );
-    jsonMapper = injector.getInstance(ObjectMapper.class);
-    properties = injector.getInstance(Properties.class);
+    JSON_MAPPER = injector.getInstance(ObjectMapper.class);
+    INDEX_IO = injector.getInstance(IndexIO.class);
+    INDEX_MERGER = injector.getInstance(IndexMerger.class);
+    INDEX_MAKER = injector.getInstance(IndexMaker.class);
```

Comment:
I wonder if it would be possible to make those fields non-static as well? This might make it easier to have configurable defaults, even for hadoop tasks.